teachapp/app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CoursesController < ApplicationController
+  def index
+    @courses = Course.all
+  end
+
+  def create
+    @course = Course.new(post_params)
+
+    if @course.save
+      # render success in Jbuilder
+    else
+      render json: { message: "400 Bad Request" }, status: :bad_request
+    end
+  end
+
+  def update
+    @course = Course.find_by_id(params[:id])
+
+    if @course.nil?
+      render json: { message: "Cannot find course" }, status: :not_found
+    else
+      @course.update(course_params)
```

Comment:
update is a function, so please check the value it returns, would be `true` if it could save it, and `false` otherwise, so you should respond to the client accordingly